packages/gestalt/src/TableRowExpandable.js

```diff
@@ -0,0 +1,51 @@
+// @flow strict
+import React, { type Node, useState } from 'react';
+import cx from 'classnames';
+import styles from './Table.css';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+
+type Props = {|
+  children: Node,
+  hoverStyle?: 'gray' | 'none',
+  expandedContents: Node,
+|};
+
+export default function TableRowExpandable(props: Props): Node {
+  const [expanded, setExpanded] = useState(false);
+  const [hovered, setHover] = useState(false);
+  const handleHover: (isHovered: boolean) => void = (isHovered: boolean) =>
+    setHover(isHovered);
+  const hoverStyle = props.hoverStyle || 'gray';
+  const cs =
+    hovered && hoverStyle === 'gray'
+      ? cx(styles.hoverShadeGray)
+      : cx(styles.hoverNone);
+
+  return (
+    <>
+      <tr
+        className={cs}
+        onMouseEnter={() => handleHover(true)}
```

Comment:
@faisalgedi we can handle this in CSS instead of having to do a mouseEnter / mouseLeave

```
.hoverShadeGray:hover {
  background-color: var(--g-colorGray150);
}
```

and no need to have `.hoverNone`, we can just leave it as an empty string:

```
const classes = hoverStyle === 'gray'
      ? cx(styles.hoverShadeGray)
      : '';
```